src/main.rs

```diff
@@ -234,6 +237,7 @@ impl Format {
         match self {
             Format::Fail => false,
             Format::Empty => true,
+            Format::EndOfInput => false,
```

Comment:
Unsure about this!

Reply:
I think `EndOfInput` is nullable? Given that it's equivalent to `Empty` with the added constraint that it must occur at the end of the input. So for example `Repeat(EndOfInput)` at the end of the file would present a problem!

Reply:
Thinking about it some more yes it's definitely nullable as `EndOfInput` can match the empty string.

Reply:
I added a doc comment that hopefully clarifies the purpose of this method some more